criu/cr-restore.c

```diff
@@ -3377,7 +3377,15 @@ static struct thread_creds_args *rst_prep_creds_args(CredsEntry *ce, unsigned lo
 	if (!args)
 		return ERR_PTR(-ENOMEM);
 
-	args->cap_last_cap = kdat.last_cap;
+	/* When restoring capability bounding set, don't attempt to mask out
+	 * caps which weren't known at the dumping side but are known at the
+	 * restore side - the dumped process couldn't have set that mask, and
+	 * the default is to not mask them out. Also in the opposite direction:
```

Comment:
>  the dumped process couldn't have set that mask, and the default is to not mask them out.

hmm, it looks more logical to drop all new caps from restored bounding sets, doesn't it?